libraries/gpu-gl-common/src/gpu/gl/GLTextureTransfer.cpp

```diff
@@ -185,7 +185,11 @@ void GLTextureTransferEngineDefault::updateMemoryPressure() {
 
     size_t allowedMemoryAllocation = gpu::Texture::getAllowedGPUMemoryUsage();
     if (0 == allowedMemoryAllocation) {
-        allowedMemoryAllocation = DEFAULT_ALLOWED_TEXTURE_MEMORY;
+        allowedMemoryAllocation = GLBackend::getTotalMemory();
```

Comment:
should this be getAvailableMemory?  and if so, what happens below if we actually DO have 0 available memory

Reply:
This is slightly unfinished, sorry about that. I rushed a bit to push this out to get some feedback and testing on other systems. This is going to be slightly more complex.

The original code just sets a limit and keeps things under that. So if we have 4 GB VRAM, we try not to exceed that amount of textures.

But systems use VRAM for other things, and some cards like NVidia and ATI let you ask how much memory is free. So if we're doing things automatically, it'd be better to actually ask how much is free, and go by that.

But for Intel so far I can only find the total memory, so we still have to do things the old way.

For now the code is 100% the old way of doing things, but the improvement's coming soon.

Reply:
Okay, improvements made in the last commit